requirements.txt

```diff
@@ -170,24 +178,28 @@ pycparser==2.21
     # via cffi
 pydyf==0.5.0
     # via weasyprint
-pyjwt[crypto]==2.5.0
```

Comment:
There are multiple dependency updates here, maybe issue them in a separate PR

Reply:
Yeah that was because of a merge, I guess.. not intended. I'll check and remove the wrong ones though.

Reply:
OK, I've reduced the number of changes here. However, I don't understand how this works.

How do I get this right? If I run `invoke setup-dev` in my repo, outside of Docker, I get updates to the pre-commit yaml file which have to be committed before I can continue. Is that expected?

Also, I keep getting changes to the header of the requirements.txt file because I have python3.10 instead of python3.9.

For now, I've forced it to be as you expected via Github - but it cost me a lot of time, and it's not something I really want to be spending time on.

Reply:
@miggland that is not expected behaviour. It could be that it changes your env the first time you run 'setup-dev' as it needs to pull the latest devtools and runs pip install for that.
Changes in the dependencies can have unforeseen consequences and as we do not have 100% coverage it is better to not do them without some manual testing/canary beforehand. This PR is probably one of the last before the new stable release so better no experiments.

Reply:
> Changes in the dependencies can have unforeseen consequences and as we do not have 100% coverage it is better to not do them without some manual testing/canary beforehand. This PR is probably one of the last before the new stable release so better no experiments.

Yes, I have no problem with that.

But how the *)(F)* do I set up my system to allow me to work without causing problems? You can't expect everyone to have exactly the same version of every package installed on their system, *outside* of Docker - if that's what you want, it should be in a controlled image.